src/swagger/specGenerator.ts

```diff
@@ -35,30 +36,111 @@ export class SpecGenerator {
+  public throwIfNotDataType(strToTest: string): Swagger.DataType {
+    const guiltyUntilInnocent = strToTest as Swagger.DataType;
+    if (
+        guiltyUntilInnocent === 'array' ||
+        guiltyUntilInnocent === 'boolean' ||
+        guiltyUntilInnocent === 'integer' ||
+        guiltyUntilInnocent === 'number' ||
+        guiltyUntilInnocent === 'object' ||
+        guiltyUntilInnocent === 'string'
+    ) {
+        return guiltyUntilInnocent;
+    } else {
+        return assertNever(guiltyUntilInnocent);
     }
+  }
 
-    if (type.dataType === 'enum') {
-      return this.getSwaggerTypeForEnumType(type as Tsoa.EnumerateType);
-    }
+  protected getSwaggerType(type: Tsoa.Type): Swagger.Schema | Swagger.BaseSchema {
 
-    return this.getSwaggerTypeForReferenceType(type as Tsoa.ReferenceType) as Swagger.Schema;
+    if (type.dataType === 'void') {
+        return this.getSwaggerTypeForVoid(type.dataType);
+    } else if (type.dataType === 'refEnum' || type.dataType === 'refObject') {
+        return this.getSwaggerTypeForReferenceType(type as Tsoa.ReferenceType);
+    } else if (
+        type.dataType === 'any' ||
+        type.dataType === 'binary' ||
+        type.dataType === 'boolean' ||
+        type.dataType === 'buffer' ||
+        type.dataType === 'byte' ||
+        type.dataType === 'date' ||
+        type.dataType === 'datetime' ||
+        type.dataType === 'double' ||
+        type.dataType === 'float' ||
+        type.dataType === 'integer' ||
+        type.dataType === 'long' ||
+        type.dataType === 'object' ||
+        type.dataType === 'string'
+    ) {
+        return this.getSwaggerTypeForPrimitiveType(type.dataType);
+    } else if (type.dataType === 'array') {
+        return this.getSwaggerTypeForArrayType(type as Tsoa.ArrayType);
+    } else if (type.dataType === 'enum') {
+        return this.getSwaggerTypeForEnumType(type as Tsoa.EnumerateType);
+    } else {
+        return assertNever(type.dataType);
+    }
   }
 
   protected getSwaggerTypeForReferenceType(referenceType: Tsoa.ReferenceType): Swagger.BaseSchema {
-    return {};
+    return {
+        // Dont' set additionalProperties value here since it will be set within the modle of the $ref when it gets created
+    };
+  }
+
+  protected getSwaggerTypeForVoid(dataType: 'void'): Swagger.BaseSchema {
+    // Described here: https://swagger.io/docs/specification/describing-responses/#empty
+    const voidSchema = {
+        // isn't allowed to have additionalProperties at all (meaning not a boolean or object)
+    };
+    return voidSchema;
   }
 
-  protected getSwaggerTypeForPrimitiveType(type: Tsoa.Type): Swagger.Schema | undefined {
-    const map = {
-      any: { type: 'object' },
+  protected getSwaggerTypeForPrimitiveType(dataType: Tsoa.PrimitiveTypeLiteral): Swagger.Schema {
+
+    const defaultAdditionalPropertiesSetting = true;
+
+    if (dataType === 'object') {
+      if (process.env.NODE_ENV !== 'tsoa_test') {
```

Comment:
why is using `object` not discouraged in tests?

Reply:
I didn’t want the test output to get cluttered with these console logs. And since we don’t use a logger with levels (such a `pino` or `Bunyan`) this is the only want to configure the log to not show up on test runs. The beauty is that thanks to us using `tsoa_test` instead of the standard `test` this should still give the message to all of the consumers of the library.

Reply:
But what I mean is.. if we generally discourage it, why is it in the tests? Or am I misunderstanding something?

Reply:
Its included in the tests because I merged this PR that a new tsoa user really wanted: https://github.com/lukeautry/tsoa/pull/378

Unfortunately after zip merged that PR, I realized that `object` doesn’t make a lot of sense. It makes even less sense when you don’t allow additional properties. I’d appreciate any guidance. What do you think we should do @endor?

Btw, here’s a great description of why object is strange in TypeScript. https://stackoverflow.com/a/18961904/706768

Reply:
I guess we'll have to live with it. It's not a big deal, just something we'll probably be wondering about after a while. It's a difficult line to draw between nudging people to use a better way while also allowing older codebases.